Error: Cannot find module '@opentelemetry/winston-transport'

<!-- 
Please answer these questions before submitting a bug report.
-->

### What version of OpenTelemetry are you using?

    "@opentelemetry/api": "1.8.0",
    "@opentelemetry/core": "1.23.0",
    "@opentelemetry/instrumentation-express": "0.37.0",
    "@opentelemetry/instrumentation-grpc": "0.50.0",
    "@opentelemetry/instrumentation-http": "0.50.0",
    "@opentelemetry/instrumentation-winston": "0.36.0",
    "@opentelemetry/resources": "1.23.0",
    "@opentelemetry/sdk-node": "0.50.0",
    "@opentelemetry/semantic-conventions": "1.23.0",

### What version of Node are you using?

18.x

### What did you do?

```js
if (process.env.Trace__Enabled === 'true') {
  const { NodeSDK } = require('@opentelemetry/sdk-node');
  const {
    HttpInstrumentation,
  } = require('@opentelemetry/instrumentation-http');
  const {
    ExpressInstrumentation,
  } = require('@opentelemetry/instrumentation-express');
  const {
    WinstonInstrumentation,
  } = require('@opentelemetry/instrumentation-winston');
  const { envDetector } = require('@opentelemetry/resources');
  const { Resource } = require('@opentelemetry/resources');
  const {
    SemanticResourceAttributes,
  } = require('@opentelemetry/semantic-conventions');
  const api = require('@opentelemetry/api');
  const core = require('@opentelemetry/core');
  const expressIgnoreLayersType =
    process.env.OTEL_EXPRESS_IGNORELAYERSTYPE || '';

  api.diag.setLogger(new api.DiagConsoleLogger(), core.getEnv().OTEL_LOG_LEVEL);

  const sdk = new NodeSDK({
    resource: new Resource({
      [SemanticResourceAttributes.SERVICE_NAME]: process.env.OTEL_SERVICE_NAME,
      [SemanticResourceAttributes.SERVICE_VERSION]:
        process.env.AppVersion,
    }),
    instrumentations: [
      new HttpInstrumentation(),
      new ExpressInstrumentation({
        ignoreLayersType: expressIgnoreLayersType.split(','),
      }),
      new WinstonInstrumentation(),
    ],
    resourceDetectors: [envDetector],
  });

  try {
    sdk.start();
    api.diag.info('OpenTelemetry instrumentation started successfully');
  } catch (error) {
    api.diag.error(
      'Error initializing OpenTelemetry SDK. Your application is not instrumented and will not produce telemetry',
      error
    );
  }
  process.on('SIGTERM', () => {
    sdk
      .shutdown()
      .then(() => api.diag.debug('OpenTelemetry SDK terminated'))
      .catch((error: unknown) =>
        api.diag.error('Error terminating OpenTelemetry SDK', error)
      );
  });
}
```

### What did you expect to see?

The app should run fine and I should see the "OpenTelemetry instrumentation started successfully" message when it is started

### What did you see instead?

```
@opentelemetry/instrumentation-winston OpenTelemetry Winston transport is not available, log records will not be automatically sent. Error: Cannot find module '@opentelemetry/winston-transport'
Require stack:
- /usr/src/app/node_modules/@opentelemetry/instrumentation-winston/build/src/instrumentation.js
- /usr/src/app/node_modules/@opentelemetry/instrumentation-winston/build/src/index.js
- /usr/src/app/apps/gateway/trace.js
```

### Additional context

I noticed that this page https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/packages/winston-transport/README.md says the following:

> This package exports the Winston transport class that is used to send records to the OpenTelemetry Logs SDK. It can be used directly when configuring a Winston logger. If using @opentelemetry/instrumenation-winston package there is no need to instantiate the transport as the instrumentation will take care of that.

But... @opentelemetry/instrumenation-winston package uses the winston-transport without even listing it in its dependencies (only in [dev deps](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/plugins/node/opentelemetry-instrumentation-winston/package.json#L63)). This is where require happens in the built code:

```
    _getPatchedConfigure() {
        return (original) => {
            const instrumentation = this;
            return function patchedConfigure(...args) {
                const config = instrumentation.getConfig();
                if (!config.disableLogSending) {
                    if (args && args.length > 0) {
                        // Try to load Winston transport
                        try {
                            const { OpenTelemetryTransportV3, } = require('@opentelemetry/winston-transport'); // fails
```

--- a/plugins/node/opentelemetry-instrumentation-winston/src/instrumentation.ts
+++ b/plugins/node/opentelemetry-instrumentation-winston/src/instrumentation.ts
@@ -221,9 +221,7 @@ export class WinstonInstrumentation extends InstrumentationBase {
               args[0].transports = newTransports;
             } catch (err) {
               instrumentation._diag.warn(
-                'OpenTelemetry Winston transport is not available, log records will not be automatically sent.',
-                err
-              );
+                '@opentelemetry/winston-transport is not available, log records will not be automatically sent.');
             }
           }
         }

--- a/plugins/node/opentelemetry-instrumentation-winston/src/instrumentation.ts
+++ b/plugins/node/opentelemetry-instrumentation-winston/src/instrumentation.ts
@@ -221,7 +221,8 @@ export class WinstonInstrumentation extends InstrumentationBase {
               args[0].transports = newTransports;
             } catch (err) {
               instrumentation._diag.warn(
-                '@opentelemetry/winston-transport is not available, log records will not be automatically sent.');
+                '@opentelemetry/winston-transport is not available, log records will not be automatically sent.'
+              );
             }
           }
         }

